File: Zad3/app/src/main/java/com/example/bmizad2/ui/notifications/PPEFragment.java
package com.example.bmizad2.ui.notifications;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import com.example.bmizad2.R;
import com.example.bmizad2.databinding.FragmentPpecalculatorBinding;

public class PPEFragment extends Fragment {
    private FragmentPpecalculatorBinding binding;
    EditText weightEditText, heightEditText, ageEditText;
    RadioGroup sexRadioGroup;
    RadioButton maleRadioButton, femaleRadioButton;
    Button calculateButton;
    TextView bmrTextView;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        PPEViewModel notificationsViewModel =
                new ViewModelProvider(this).get(PPEViewModel.class);

        binding = FragmentPpecalculatorBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        weightEditText = root.findViewById(R.id.weightEditText);
        heightEditText = root.findViewById(R.id.heightEditText);
        ageEditText = root.findViewById(R.id.ageEditText);
        sexRadioGroup = root.findViewById(R.id.sexRadioGroup);
        maleRadioButton = root.findViewById(R.id.maleRadioButton);
        femaleRadioButton = root.findViewById(R.id.femaleRadioButton);
        calculateButton = root.findViewById(R.id.calculateButton);
        bmrTextView = root.findViewById(R.id.bmrTextView);

        calculateButton.setOnClickListener(view -> {
            double weight = Double.parseDouble(weightEditText.getText().toString());
            double height = Double.parseDouble(heightEditText.getText().toString());
            int age = Integer.parseInt(ageEditText.getText().toString());
            int sex = 0;
            if (maleRadioButton.isChecked()) {
                sex = 1;
            } else if (femaleRadioButton.isChecked()) {
                sex = 2;
            }
            double ppm = calculatePPM(weight, height, age, sex);
            weightEditText.getText().clear();
            heightEditText.getText().clear();
            ageEditText.getText().clear();
            sexRadioGroup.clearCheck();

            bmrTextView.setText(String.format("Your PPM is %.2f", ppm));

        });

        final TextView textView = binding.textPpecalculator;
        notificationsViewModel.getText().observe(getViewLifecycleOwner(), textView::setText);
        return root;
    }

    private double calculatePPM(double weight, double height, int age, int sex) {
        switch (sex) {
            case 2:
                return 655.1f + (9.563f * weight) + (1.85f * height) - (4.676f * age);
            case 1:
                return 66.5f + (13.75f * weight) + (5.003f * height) - (6.775f * age);
        }
        return 0;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

}

File: Zad4/app/src/main/java/com/example/bmizad2/ui/notifications/PPEFragment.kt
package com.example.bmizad2.ui.notifications

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.example.bmizad2.R
import com.example.bmizad2.databinding.FragmentPpecalculatorBinding

class PPEFragment : Fragment() {
    private var binding: FragmentPpecalculatorBinding? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {
        val notificationsViewModel = ViewModelProvider(this).get(
            PPEViewModel::class.java
        )
        binding = FragmentPpecalculatorBinding.inflate(inflater, container, false)

        val root: View = binding?.root ?: return null

        val weightEditText = root.findViewById<EditText>(R.id.weightEditText)
        val heightEditText = root.findViewById<EditText>(R.id.heightEditText)
        val ageEditText = root.findViewById<EditText>(R.id.ageEditText)
        val sexRadioGroup = root.findViewById<RadioGroup>(R.id.sexRadioGroup)
        val maleRadioButton = root.findViewById<RadioButton>(R.id.maleRadioButton)
        val femaleRadioButton = root.findViewById<RadioButton>(R.id.femaleRadioButton)
        val calculateButton = root.findViewById<Button>(R.id.calculateButton)
        val bmrTextView = root.findViewById<TextView>(R.id.bmrTextView)

        calculateButton.setOnClickListener {
            val weight = weightEditText.text.toString().toDoubleOrNull() ?: 0.0
            val height = heightEditText.text.toString().toDoubleOrNull() ?: 0.0
            val age = ageEditText.text.toString().toIntOrNull() ?: 0
            var sex = 0
            if (maleRadioButton.isChecked) {
                sex = 1
            } else if (femaleRadioButton.isChecked) {
                sex = 2
            }
            val ppm = calculatePPM(weight, height, age, sex)
            weightEditText.text.clear()
            heightEditText.text.clear()
            ageEditText.text.clear()
            sexRadioGroup.clearCheck()
            bmrTextView.text = String.format("Your PPM is %.2f", ppm)
        }

        notificationsViewModel.text.observe(viewLifecycleOwner) { text ->
            binding?.textPpecalculator?.text = text
        }

        return root
    }

    private fun calculatePPM(weight: Double, height: Double, age: Int, sex: Int): Double {
        return when (sex) {
            2 -> 655.1 + 9.563 * weight + 1.85 * height - 4.676 * age
            1 -> 66.5 + 13.75 * weight + 5.003 * height - 6.775 * age
            else -> 0.0
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }
}

